logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -181,8 +182,22 @@ private void ensureNotStarted() {
         }
     }
 
+    /**
+     * It is possible that requestContextStorageProvider hasn't been initialized yet
+     * due to static variable circular dependency.
+     * Most notably, this can happen when logs are appended while trying to initialize
+     * Flags#requestContextStorageProvider.
+     */
+    private static boolean isInitialized() {
+        return Flags.initialized();
+    }
+
     @Override
     protected void append(ILoggingEvent eventObject) {
+        if (!isInitialized()) {
+            return;
+        }
+
         if (exporter == null) {
             exporter = builder.build();
         }
```

Comment:
Should we defer the creation of `exporter` until `isInitialized()` is true?

Reply:
; so that `isInitialized()` is only invoked when `exporter` is null.

Reply:
I'm unsure what this means 😅
Doesn't the current change defer setting a value to `exporter` until `isInitialized() == true` by returning early already?